modules/chat/src/routes/routes.ts

```diff
@@ -72,23 +92,84 @@ export class ChatRoutes {
     }
 
     try {
-      await validateUsersInput(this.grpcSdk, users);
+      usersToBeAdded = await validateUsersInput(this.grpcSdk, users);
     } catch (e) {
       throw new GrpcError(status.INTERNAL, e.message);
     }
 
-    room.participants = Array.from(new Set([...room.participants, ...users]));
-    await ChatRoom.getInstance()
-      .findByIdAndUpdate(
-        room._id,
-        room,
-      )
-      .catch((e: Error) => {
-        throw new GrpcError(status.INTERNAL, e.message);
+    const config = await this.grpcSdk.config.get('chat');
+    let retMessage: any [] = [];
+    if (config.sendInvitations.enabled) {
+      const serverConfig = await this.grpcSdk.config.getServerConfig();
+      const url = serverConfig.url;
+      for (const invitedUser of usersToBeAdded) {
+        const invitationsCount = await InvitationToken.getInstance().countDocuments({
+          room: roomId,
+          receiver: invitedUser._id,
+          sender: user._id,
+        });
+        if (room.participants.includes(invitedUser._id)) {
```

Comment:
No need to complete the loop.
Both checks should immediately throw an error after finding a single existing member or pre-invited user to avoid wasting cycles on a failed request.

eg `throw new GrpcError(status.INVALID_ARGUMENT, 'users array contains existing member or invited user ids');`